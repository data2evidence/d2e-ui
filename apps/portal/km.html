<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8" />
    <title>Kaplan-Meier Plot with ECharts</title>
    <script src="https://cdn.jsdelivr.net/npm/echarts/dist/echarts.min.js"></script>
  </head>
  <body>
    <div id="main" style="width: 800px; height: 600px"></div>
    <script>
      var data = {
        timeX: [
          0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29,
          30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56,
          57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83,
          84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108,
          109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130,
          131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152,
          153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174,
          175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 187, 188, 189, 190, 191, 192, 193, 194, 195, 196,
          197, 198, 199, 200, 201, 202, 203, 204, 205, 206, 207, 208, 209, 210, 211, 212, 213, 214, 215, 216, 217, 218,
          219, 220, 221, 222, 223, 224, 225, 226, 227, 228, 229, 230, 231, 232, 233, 234, 235, 236, 237, 238, 239, 240,
          241, 242, 243, 244, 245, 246, 247, 248, 249, 250, 251, 252, 253, 254, 255, 256, 257, 258, 259, 260, 261, 262,
          263, 264, 265, 266, 267, 268, 269, 270, 271, 272, 273, 274, 275, 276, 277, 278, 279, 280, 281, 282, 283, 284,
          285, 286, 287, 288, 289, 290, 291, 292, 293, 294, 295, 296, 297, 298, 299, 300, 301, 302, 303, 304, 305, 306,
          307, 308, 309, 310, 311, 312, 313, 314, 315, 316, 317, 318, 319, 320, 321, 322, 323, 324, 325, 326, 327, 328,
          329, 330, 331, 332, 333, 334, 335, 336, 337, 338, 339, 340, 341, 342, 343, 344, 345, 346, 347, 348, 349, 350,
          351, 352, 353, 354, 355, 356, 357, 358, 359, 360, 361, 362, 363, 364, 365, 366, 367, 368, 369, 370, 371, 372,
          373, 374, 375, 376, 377, 378, 379, 380, 381, 382, 383, 384, 385, 386, 387, 388, 389, 390, 391, 392, 393, 394,
          395, 396, 397, 398, 399, 400, 401, 402, 403, 404, 405, 406, 407, 408, 409, 410, 411, 412, 413, 414, 415, 416,
          417, 418, 419, 420, 421, 422, 423, 424,
        ],
        survivalY: [
          1, 0.9697, 0.9494, 0.9386, 0.927, 0.9198, 0.9133, 0.9089, 0.8995, 0.893, 0.8858, 0.8793, 0.8749, 0.8713,
          0.8691, 0.8648, 0.8612, 0.8561, 0.8518, 0.8489, 0.8424, 0.8395, 0.8315, 0.8293, 0.8235, 0.8185, 0.8149,
          0.8091, 0.804, 0.7997, 0.7939, 0.7888, 0.783, 0.7801, 0.7758, 0.7707, 0.7671, 0.7627, 0.7584, 0.754, 0.7504,
          0.7482, 0.7424, 0.7373, 0.7323, 0.7286, 0.7177, 0.7156, 0.709, 0.7025, 0.6967, 0.6959, 0.6908, 0.685, 0.6806,
          0.677, 0.6748, 0.6718, 0.6682, 0.6652, 0.6615, 0.6563, 0.6533, 0.6488, 0.6428, 0.639, 0.6322, 0.6277, 0.6246,
          0.6201, 0.6155, 0.6071, 0.6055, 0.5993, 0.5962, 0.5924, 0.5861, 0.5791, 0.5752, 0.5728, 0.5697, 0.5641,
          0.5609, 0.5593, 0.5529, 0.5472, 0.5456, 0.5415, 0.5374, 0.535, 0.5317, 0.5284, 0.5251, 0.5184, 0.5133, 0.5116,
          0.5074, 0.5005, 0.4945, 0.4885, 0.485, 0.4824, 0.4789, 0.4718, 0.4691, 0.4673, 0.4637, 0.4565, 0.4538, 0.451,
          0.4446, 0.441, 0.4382, 0.4354, 0.4317, 0.4279, 0.4232, 0.4204, 0.4185, 0.4176, 0.4156, 0.4118, 0.4071, 0.4061,
          0.4022, 0.3993, 0.3954, 0.3945, 0.3915, 0.3876, 0.3856, 0.3856, 0.3856, 0.3816, 0.3786, 0.3735, 0.3674,
          0.3633, 0.3613, 0.3561, 0.3509, 0.3489, 0.3489, 0.3478, 0.3447, 0.3404, 0.3372, 0.3329, 0.3318, 0.3296,
          0.3285, 0.3252, 0.3219, 0.3196, 0.3185, 0.3151, 0.3129, 0.3106, 0.3038, 0.3038, 0.3038, 0.3026, 0.3003,
          0.2991, 0.2979, 0.2955, 0.293, 0.2905, 0.2893, 0.2868, 0.2842, 0.2803, 0.2777, 0.2739, 0.2725, 0.2699, 0.2672,
          0.2672, 0.2645, 0.2604, 0.259, 0.259, 0.2576, 0.2561, 0.2547, 0.2533, 0.2533, 0.2518, 0.2518, 0.2487, 0.2487,
          0.244, 0.2424, 0.2407, 0.2407, 0.2407, 0.2373, 0.2373, 0.2373, 0.2373, 0.2373, 0.2336, 0.2317, 0.2317, 0.2317,
          0.2317, 0.2298, 0.2298, 0.2258, 0.2258, 0.2258, 0.2258, 0.2236, 0.2215, 0.2192, 0.2169, 0.2146, 0.2146,
          0.2146, 0.2146, 0.2122, 0.2122, 0.2073, 0.2048, 0.2048, 0.2048, 0.2048, 0.2048, 0.1961, 0.1961, 0.1961,
          0.1961, 0.1961, 0.1931, 0.1931, 0.19, 0.19, 0.19, 0.19, 0.19, 0.1869, 0.1869, 0.1869, 0.1869, 0.1869, 0.1869,
          0.1869, 0.1834, 0.1834, 0.1834, 0.1798, 0.1798, 0.176, 0.176, 0.176, 0.172, 0.168, 0.164, 0.16, 0.16, 0.1516,
          0.1516, 0.1516, 0.1516, 0.1516, 0.1427, 0.1427, 0.1427, 0.1427, 0.1427, 0.1427, 0.1427, 0.1379, 0.1379,
          0.1379, 0.1379, 0.1379, 0.1379, 0.133, 0.133, 0.133, 0.133, 0.1277, 0.1277, 0.1277, 0.1277, 0.1277, 0.1216,
          0.1216, 0.1216, 0.1216, 0.1216, 0.1216, 0.1216, 0.1216, 0.1216, 0.1216, 0.1216, 0.1216, 0.1216, 0.1216,
          0.1216, 0.1216, 0.1216, 0.1216, 0.1216, 0.1216, 0.1216, 0.1216, 0.1216, 0.1216, 0.1216, 0.1216, 0.1216,
          0.1129, 0.1026, 0.1026, 0.1026, 0.1026, 0.1026, 0.1026, 0.0912, 0.0912, 0.0912, 0.0912, 0.0912, 0.0912,
          0.0912, 0.0912, 0.0912, 0.0912, 0.0912, 0.0912, 0.0912, 0.0912, 0.0912, 0.0912, 0.0912, 0.0912, 0.0912,
          0.0912, 0.0912, 0.0912, 0.0912, 0.0912, 0.0912, 0.0912, 0.0912, 0.0912, 0.0912, 0.0912, 0.0912, 0.0912,
          0.0912, 0.0912, 0.0912, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684,
          0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684,
          0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684,
          0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684,
          0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684,
          0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0.0684, 0,
        ],
      };
      // Initialize the chart
      var myChart = echarts.init(document.getElementById("main"));

      // Data for confidence intervals
      var lowerCI = data.survivalY.map((d) => d * 0.95);
      var upperCI = data.survivalY.map((d) => d * 1.05);

      // Transform the data for step plot and confidence intervals
      var times = [];
      var survivals = [];
      var lowerBounds = [];
      var upperBounds = [];
      for (var i = 0; i < data.survivalY.length; i++) {
        times.push(data.timeX[i]);
        survivals.push(data.survivalY[i]);
        // lowerBounds.push(lowerCI[i]);
        // upperBounds.push(upperCI[i]);
        if (i < data.survivalY.length - 1) {
          times.push(data.timeX[i + 1]);
          survivals.push(data.survivalY[i]);
          // lowerBounds.push(lowerCI[i]);
          // upperBounds.push(upperCI[i]);
        }
      }

      // Chart options
      var option = {
        title: {
          text: "Kaplan-Meier Survival Curve",
        },
        tooltip: {
          trigger: "axis",
        },
        xAxis: {
          type: "value",
          name: "Time",
        },
        yAxis: {
          type: "value",
          name: "Survival Probability",
        },
        tooltip: {
          trigger: "axis",
          formatter: function (params) {
            var result = params[0].axisValueLabel + "<br>";
            var displayed = new Set(); // To keep track of displayed survival probabilities
            params.forEach(function (item) {
              if (item.seriesName === "Survival Probability" && !displayed.has(item.data[1])) {
                result += item.marker + item.seriesName + ": " + item.data[1] + "<br>";
                displayed.add(item.data[1]); // Mark the survival probability as displayed
              }
            });
            return result;
          },
        },
        series: [
          {
            name: "Survival Probability",
            data: times.map((time, index) => [time, survivals[index]]),
            type: "line",
            step: "end",
            smooth: true,
            lineStyle: {
              color: "black",
            },
          },
          {
            name: "Lower CI",
            data: times.map((time, index) => [time, lowerBounds[index]]),
            type: "line",
            step: "end",
            lineStyle: {
              opacity: 0,
            },
            areaStyle: {
              color: "rgba(0, 0, 0, 0.2)",
              origin: "start",
            },
          },
          {
            name: "Upper CI",
            data: times.map((time, index) => [time, upperBounds[index]]),
            type: "line",
            step: "end",
            lineStyle: {
              opacity: 0,
            },
            areaStyle: {
              color: "rgba(0, 0, 0, 0.2)",
              origin: "end",
            },
          },
        ],
      };

      // Use the specified configuration and data to show the chart.
      myChart.setOption(option);
    </script>
  </body>
</html>
