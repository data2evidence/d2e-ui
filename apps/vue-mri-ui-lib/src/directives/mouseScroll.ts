export default {
  mounted(el, { value }: { value?: (event: any) => void }) {
    if (!value) {
      return
    }
    const wheelEvent = event => {
      value(event)
    }
    el.addEventListener('mousewheel', wheelEvent)
    el.addEventListener('DOMMouseScroll', wheelEvent)
  },
}
