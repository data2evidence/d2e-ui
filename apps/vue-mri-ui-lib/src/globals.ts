;(window as any).sap = (window as any).sap || {
  hc: {
    mri: {
      pa: {
        ui: {
          Utils: {},
        },
      },
    },
  },
  ui: {
    getCore: () => ({
      byId: () => text => 'ABC',
      getEventBus: () => ({
        subscribe:
          () =>
          (text, {}) => ({}),
        unsubscribe:
          () =>
          (text, {}) => ({}),
      }),
    }),
    model: {
      resource: {
        ResourceModel: () => ({
          getResourceBundle: () => ({
            getText: () => text => 'ABC',
          }),
          _oResourceBundle: {
            sLocale: 'ABC',
          },
        }),
      },
    },
    require: () => ({}),
  },
}
export {}
