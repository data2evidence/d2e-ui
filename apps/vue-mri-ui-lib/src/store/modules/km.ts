import * as types from '../mutation-types'

// initial state
const state = {
  series: [],
  categories: [],
  ticks: [],
  tickType: 'years',
  ticksData: [],
  displayInfo: {},
  firstLoad: {},
}

const getters = {
  getKMCategories: modulestate => modulestate.categories,
  getKMSeries: modulestate => modulestate.series,
  getTicks: modulestate => modulestate.ticks,
  getTickType: modulestate => modulestate.tickType,
  getTicksData: modulestate => modulestate.ticksData,
  getKMDisplayInfo: modulestate => modulestate.displayInfo,
  getKMFirstLoad: modulestate => modulestate.firstLoad,
}

// actions
const actions = {
  setKMLegends({ commit }, { categories, series }) {
    commit(types.SET_KM_LEGENDS, { categories, series })
  },
  setTicks({ commit }, { kmTicks, kmTickType }) {
    commit(types.SET_KM_TICKS, { kmTicks, kmTickType })
  },
  setTicksData({ commit }, { kmTicksData }) {
    commit(types.SET_KM_TICKS_DATA, { kmTicksData })
  },
  setKMDisplayInfo({ commit }, { displayInfo }) {
    commit(types.SET_KM_DISPLAY_INFO, { displayInfo })
  },
  setKMFirstLoad({ commit }, { firstLoad }) {
    commit(types.SET_KM_FIRST_LOAD, { firstLoad })
  },
}

// mutations
const mutations = {
  [types.SET_KM_LEGENDS](modulestate, { categories, series }) {
    modulestate.categories = categories
    modulestate.series = series
  },
  [types.SET_KM_TICKS](modulestate, { kmTicks, kmTickType }) {
    modulestate.ticks = kmTicks
    modulestate.tickType = kmTickType
  },
  [types.SET_KM_TICKS_DATA](modulestate, { kmTicksData }) {
    modulestate.ticksData = kmTicksData
  },
  [types.SET_KM_DISPLAY_INFO](modulestate, { displayInfo }) {
    modulestate.displayInfo = displayInfo
  },
  [types.SET_KM_FIRST_LOAD](modulestate, { firstLoad }) {
    modulestate.firstLoad = firstLoad
  },
}

export default {
  state,
  getters,
  actions,
  mutations,
}
