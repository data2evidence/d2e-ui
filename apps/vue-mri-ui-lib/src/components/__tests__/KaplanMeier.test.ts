import { shallowMount } from '@vue/test-utils'
import { createStore } from 'vuex'
import '../../globals'
import km from '../KaplanMeier.vue'
import mouseScroll from '@/directives/mouseScroll'

describe('KaplanMeier.vue', () => {
  let store
  let actions
  let getters

  beforeEach(() => {
    actions = {
      disableAllAxesandProperties: jest.fn(),
      setAxisValue: jest.fn(),
      setFireRequest: jest.fn(),
      setChartPropertyValue: jest.fn(),
    }
    getters = {
      getKMDisplayInfo: () => () => {
        return { censoring: true, errorlines: 1 }
      },
      getText: (modulestate, modulegetters) => (key, param) => {
        return key
      },
      getSplitterWidth: () => {
        return {}
      },
      getChartSize: () => {
        return {}
      },
      getCsvFireDownload: () => {
        return {}
      },
      getFireRequest: () => {
        return {}
      },
    }
    store = createStore({
      actions,
      getters,
    })
  })

  it('sets the correct default data', () => {
    const wrapper = shallowMount(km as any, {
      global: {
        plugins: [store],
        directives: { 'mouse-scroll': mouseScroll },
      },
    })
    const defaultData = wrapper.vm.$data
    expect(defaultData).toMatchSnapshot()
  })
})
