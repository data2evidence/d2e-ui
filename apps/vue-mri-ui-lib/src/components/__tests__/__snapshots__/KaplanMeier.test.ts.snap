// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KaplanMeier.vue sets the correct default data 1`] = `
Object {
  "allCurves": Object {},
  "cachedRangeSliderValue": Array [
    0,
    0,
  ],
  "chartData": Object {},
  "errorMessage": "",
  "globalPValue": "",
  "kmChartStyle": Object {},
  "kmInterval": "",
  "kmLogRankHelpArrowPosition": Object {
    "left": "270px",
  },
  "kmLogRankHelpPopoverPosition": Object {
    "right": "8px",
  },
  "kmUnit": "years",
  "maxDataday": 0,
  "maxDay": 0,
  "minDay": 0,
  "minProb": 0,
  "rangeSliderValue": Array [
    0,
    0,
  ],
  "selection": Array [],
  "series": Array [],
  "showCensoring": false,
  "showErrorLines": false,
  "showKMLogRankHelpPopOver": false,
  "showKMLogRankHelpPopOverHover": false,
  "showKMLogRankHelpPopOverIcon": false,
  "showKMStatisticsErrPopupForMoreCurves": false,
  "showKMStatisticsErrPopupForOneCurve": false,
  "showKMStatisticsPopup": false,
  "showSubComponents": false,
  "showTooltip": false,
  "sliderFocus": false,
  "sliderHoverState": false,
  "sliderWidth": 0,
  "tooltipCategories": Array [],
  "tooltipFocusLeft": false,
  "tooltipFocusRight": false,
  "tooltipHoverState": false,
  "tooltipMeasures": Array [],
  "tooltipPosition": Object {},
  "xScale": 0,
  "yScale": 0,
}
`;
